packages/core/src/model.ts

```diff
@@ -454,6 +465,46 @@ export class ChartModel {
 		return this.colorScale;
 	}
 
+	isCustomColorValid() {
+		const userProvidedScale = Tools.getProperty(
+			this.getOptions(),
+			"color",
+			"scale"
+		);
+		const dataGroups = this.getDataGroups();
+
+		if (
+			userProvidedScale == null ||
+			Object.keys(userProvidedScale).length < dataGroups.length
```

Comment:
what if I'm using the color palette and I provide a color scale that changes just 1 color? Would that be disregarded here?

Reply:
Yes, if user only provides partial color scale, those colors will be assigned to the components repeatedly. Currently, we only apply colors either by class or color fill. I can make another PR later for this so user can partially override the color.